src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,203 @@
+package duke;
+
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+import static java.lang.Integer.parseInt;
+
+/**
+ * Represents the main logic of the code when the user enters an input.
+ */
+public class Parser {
+    private Scanner scan;
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+    private String arr[];
+    private String input = "";
+
+    /**
+     * A constructor to create a parser object.
+     *
+     * @param scan A scanner that takes in the input from the user.
+     * @param storage A storage that deals with loading and saving files.
+     * @param taskList A list that keeps track of all tasks.
+     * @param ui An object that perform tasks related to interactions with user.
+     */
+    Parser(Scanner scan, Storage storage, TaskList taskList, Ui ui) {
+        this.scan = scan;
+        this.storage = storage;
+        this.taskList = taskList;
+        this.ui = ui;
+    }
+
+    /**
+     * Checks if a string is an integer or not.
+     *
+     * @param s A string from the user input.
+     * @return True if it is an integer, false otherwise.
+     */
+    public static boolean isNumeric(String s) {
+        try {
+            Integer.parseInt(s);
+        } catch (NumberFormatException e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Makes sense of the user input.
+     */
+    public void parse() {
+        do {
+            try {
+                input = scan.nextLine();
+                arr = input.split(" ");
+                if (input.equals("bye")) {
+                    break;
+                } else if (arr[0].equals("done")) {
+                    parseDone(arr);
+                } else if (arr[0].equals("delete")) {
+                    parseDelete(arr);
+                } else if (arr[0].equals("todo")) {
+                    parseTodo(arr);
+                } else if (arr[0].equals("deadline")) {
+                    parseDeadline(arr);
+                } else if (arr[0].equals("event")) {
+                    parseEvent(arr);
+                } else if (input.equals("list")) {
+                    parseList();
+                } else if (arr[0].equals("find")) {
+                    parseFind(arr);
+                } else {
+                    throw new InvalidCommandException("Command not Found");
+                }
+                storage.saveFile(taskList);
+            } catch (InvalidCommandException e) {
+                System.out.println(e.toString());
+            } catch (EmptyDescriptionException e) {
+                System.out.println(e.toString());
+            } catch (InvalidValueException e) {
+                System.out.println(e.toString());
+            } catch (DateTimeParseException e) {
+                System.out.println(e.getMessage());
+            }
```

Comment:
code duplication. can consider creating a parent class DukeException which all the other exception classes inherit from 
```suggestion
            } catch (DukeException e) {
                System.out.println(e.toString());
            } 
```